### ISSUE TITLE
Using a local eslint makes node load a shell script

### ISSUE THREAD

**Issue by Dinnerbone** [TheSavior/ESLint-Formatter#18]
Running windows 10, sublime text 3, ESLint-Formatter 2.1.1.

I just created an empty project and installed eslint locally. I have auto-format turned on on ESLint-Formatter config.

Seemingly 1 in 5 saves, eslint tries to load from the local eslint, and the other 4 it uses global. I assume this is probably #17 causing it to not always find the local eslint.

Whenever it uses the global eslint, it loads node with `%APPDATA%/npm/node_modules/eslint/bin/eslint` which is a Javascript file, and works great. However...

Whenever it uses the local eslint, it loads node with `node_modules/.bin/eslint` which is a shell file, and causes this to pop up as an error message:

```
Unexpected error(<class 'Exception'>): Error: <project>\node_modules\.bin\eslint:2
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")
          ^^^^^^^
SyntaxError: missing ) after argument list
    at Object.exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:513:28)
    at Object.Module._extensions..js (module.js:550:10)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.runMain (module.js:575:10)
    at run (node.js:348:7)
    at startup (node.js:140:9)
    at node.js:463:3
```

I'd suggest looking for `node_modules/eslint/bin/eslint` for local installations instead.

**Comment by TheSavior** [comment]
#17 hasn't been released yet. 

This is an interesting issue. I wonder why calling `node_modules/.bin/eslint` would cause errors. It should just be a symlink to the file in `node_modules/eslint/bin`:

```
```

**Comment by Dinnerbone** [comment]
For me, on windows, it's a shell script:

```
#!/bin/sh
basedir=$(dirname "$(echo "$0" | sed -e 's,\\,/,g')")

case `uname` in
    *CYGWIN*) basedir=`cygpath -w "$basedir"`;;
esac

if [ -x "$basedir/node" ]; then
  "$basedir/node"  "$basedir/../eslint/bin/eslint.js" "$@"
  ret=$?
else 
  node  "$basedir/../eslint/bin/eslint.js" "$@"
  ret=$?
fi
exit $ret
```

**Comment by TheSavior** [comment]
Look at that, I just learned something new about npm on windows. 

I'd happily accept a pull request for this case. I unfortunately don't have any easy way to test and validate that functionality.

**Comment by Dinnerbone** [comment]
I mean, I'm not sure why on windows it makes a shell file of all things, but hey!

I'll try changing it to search for `node_modules/eslint/bin/eslint` instead of `node_modules/.bin/eslint` and see if it works for me.

**Comment by TheSavior** [comment]
I'm thinking we'd likely want to continue running the normal `.bin/eslint` file on non-windows platforms. I'm kinda leaning towards the "If it aint broke, don't fix it" mentality. Especially since this repo doesn't have tests.

**Comment by Dinnerbone** [comment]
The default file on windows is  `eslint/bin/eslint`, but the other OSes go to the generic bin folder.

Do you want me to make a pull request that duplicates the `eslint_path` setting for local search paths? I imagine it'd look like:
```json
  // The location to search for a locally installed eslint package
  // If this is not found, it will try to use the global package (see 'eslint_path')
  "local_eslint_path": {
    "windows": "./node_modules/eslint/bin/eslint",
    "linux": "./node_modules/.bin/eslint",
    "osx": "./node_modules/.bin/eslint"
  },
```

**Comment by TheSavior** [comment]
Yeah, that seems like a reasonable approach, thanks.
